src/check_result.jl

```diff
@@ -40,12 +40,17 @@ for (T1, T2) in
 end
 
 test_approx(::AbstractZero, x, msg=""; kwargs...) = test_approx(zero(x), x, msg; kwargs...)
-test_approx(::AbstractZero, x::AbstractArray{<:AbstractArray}, msg=""; kwargs...) = test_approx(map(zero, x), x, msg; kwargs...)
 test_approx(x, ::AbstractZero, msg=""; kwargs...) = test_approx(x, zero(x), msg; kwargs...)
-test_approx(x::AbstractArray{<:AbstractArray}, ::AbstractZero, msg=""; kwargs...) = test_approx(x, map(zero, x), msg; kwargs...)
 test_approx(x::ZeroTangent, y::ZeroTangent, msg=""; kwargs...) = @test true
 test_approx(x::NoTangent, y::NoTangent, msg=""; kwargs...) = @test true
 
+function test_approx(z::AbstractZero, x::AbstractArray{<:AbstractArray}, msg=""; kwargs...)
+    for el in x
+        test_approx(el, z, msg; kwargs...)
```

Comment:
Doesn't matter but arguably a bit more consistent would be

```suggestion
        test_approx(z, el, msg; kwargs...)
```